source/_posts/2016-09-14-what_the_tag_series_ep1_document_object.markdown

```diff
@@ -0,0 +1,108 @@
+
+`document.readyState`
+
+This readonly attribute is a very important and useful one, it holds the current state of the document, it can be in three different states:
+
+**loading**
+When the document has not finished loading (the response from the server is not yet completed).
+
+**interactive**
+In case the first load is done and the parsing process is completed but there are some resources that need to be loaded, for example images, styles, javascript, etc.
+
+**complete**
+When all related resources are loaded. Once this state is achieved the `load` event will be fired.
+
+Every time the value of this attribute changes the `readystatechange` event is fired. This event is equivalent to the more know ones `load` and `DOMContentLoaded`, next is an example, that I borrowed from the docs of [Mozilla](https://developer.mozilla.org/en-US/docs/Web/API/Document/readyState) (if you need more cool examples just visit them!), this simply listens to the event and does some application initialization:
+
+```js
+// alternative to DOMContentLoaded event
+document.onreadystatechange = function () {
+  if (document.readyState === "interactive") {
+    initApplication();
+  }
+}
+```
+
+## DOM tree accessors
+
+This accessors let you interact directly with the document, by setting the values of this properties you can modify the document and change it to your will, but remember, as uncle Ben once said:
+> With great power comes great responsibility.
+
+`document.head`
+
+This returns the head element, and it is a readonly attribute. The head element is where the metadata of the document is, for example the title, styles and scripts.
+
+`document.title`
+
+This attribute is a very simple one, it lets you get the title of the document or set it. And yes, the title is the thing is written in the browser tab.
+
+`document.body`
+
+Here is where probably most of your website's HTML is, here you can access the code directly and change it. I invite you to try to break this website, investigate about `innerHTML`, (in case you succeed just refresh the site).
+
+`document.images`
+
+Using this you can get a list of all images in the document, but it is read only, it is just an easy way to access this kind of elements, you can affect this indirectly by modifying the document and adding more images. There are more than this accessor in the list take a look at the [definition of the images](https://html.spec.whatwg.org/multipage/dom.html#dom-document-images).
+
+`document.getElementsByName(name)`
+
+It queries the document and returns a list of elements where the attribute name is set to *name*. So if you have something like the code below and you do `document.getElementsByName('up')` you will get only the form element, just as simple as that!
+
+{% codeblock lang:html %}
+{% raw %}
+  <form name="up"><input type="text"></form>
+  <div name="down"><input type="text"></div>
```

Comment:
The `name` attribute is not valid in a `div` element, according [this](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes), so I would change to use an example with a different HTML markup.

Maybe something like the following:

```
<form>
  <input type="checkbox" name="up">
  <input type="checkbox" name="down">
</form>
```